{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[],"authorship_tag":"ABX9TyMlMsMBTabRhcJY/npO9leI"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"markdown","source":["# Emergence of Linguistic Communication from Referential Games with Symbolic and Pixel Input\n","\n","https://github.com/nickleomartin/emergent_comm_rl?tab=readme-ov-file"],"metadata":{"id":"_eLiiFNQk1GG"}},{"cell_type":"code","execution_count":2,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"E2uARdoQkvmo","executionInfo":{"status":"ok","timestamp":1740092218716,"user_tz":360,"elapsed":3238,"user":{"displayName":"Alan Mauricio Camargo Hernandez","userId":"09753390928466343821"}},"outputId":"6db85b38-62e0-4d38-db9d-e0aad2d6db28"},"outputs":[{"output_type":"stream","name":"stdout","text":["Cloning into 'emergent_comm_rl'...\n","remote: Enumerating objects: 361, done.\u001b[K\n","remote: Total 361 (delta 0), reused 0 (delta 0), pack-reused 361 (from 1)\u001b[K\n","Receiving objects: 100% (361/361), 151.05 KiB | 836.00 KiB/s, done.\n","Resolving deltas: 100% (233/233), done.\n","Filtering content: 100% (16/16), 201.33 KiB | 123.00 KiB/s, done.\n"]}],"source":["!git clone https://github.com/nickleomartin/emergent_comm_rl.git"]},{"cell_type":"code","source":["%cd emergent_comm_rl"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"DEgkAKJElG7F","executionInfo":{"status":"ok","timestamp":1740092249754,"user_tz":360,"elapsed":10,"user":{"displayName":"Alan Mauricio Camargo Hernandez","userId":"09753390928466343821"}},"outputId":"ece2cd70-5219-4b64-9af8-db236ac42e3d"},"execution_count":4,"outputs":[{"output_type":"stream","name":"stdout","text":["/content/emergent_comm_rl\n"]}]},{"cell_type":"code","source":["!pip install torch==1.8.1\n","!pip install -r requirements.txt"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"yWnJ4pI2kykt","executionInfo":{"status":"ok","timestamp":1740092352188,"user_tz":360,"elapsed":3134,"user":{"displayName":"Alan Mauricio Camargo Hernandez","userId":"09753390928466343821"}},"outputId":"05ee9615-a4c8-45f1-9775-14451e7340f9"},"execution_count":8,"outputs":[{"output_type":"stream","name":"stdout","text":["\u001b[31mERROR: Could not find a version that satisfies the requirement torch==1.8.1 (from versions: 1.13.0, 1.13.1, 2.0.0, 2.0.1, 2.1.0, 2.1.1, 2.1.2, 2.2.0, 2.2.1, 2.2.2, 2.3.0, 2.3.1, 2.4.0, 2.4.1, 2.5.0, 2.5.1, 2.6.0)\u001b[0m\u001b[31m\n","\u001b[0m\u001b[31mERROR: No matching distribution found for torch==1.8.1\u001b[0m\u001b[31m\n","\u001b[0mCollecting absl-py==0.2.2 (from -r requirements.txt (line 1))\n","  Using cached absl-py-0.2.2.tar.gz (82 kB)\n","  \u001b[1;31merror\u001b[0m: \u001b[1msubprocess-exited-with-error\u001b[0m\n","  \n","  \u001b[31m×\u001b[0m \u001b[32mpython setup.py egg_info\u001b[0m did not run successfully.\n","  \u001b[31m│\u001b[0m exit code: \u001b[1;36m1\u001b[0m\n","  \u001b[31m╰─>\u001b[0m See above for output.\n","  \n","  \u001b[1;35mnote\u001b[0m: This error originates from a subprocess, and is likely not a problem with pip.\n","  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25herror\n","\u001b[1;31merror\u001b[0m: \u001b[1mmetadata-generation-failed\u001b[0m\n","\n","\u001b[31m×\u001b[0m Encountered error while generating package metadata.\n","\u001b[31m╰─>\u001b[0m See above for output.\n","\n","\u001b[1;35mnote\u001b[0m: This is an issue with the package mentioned above, not pip.\n","\u001b[1;36mhint\u001b[0m: See above for details.\n"]}]},{"cell_type":"code","source":["!pip install keras\n","# !pip install np_utils"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"KT4-1h63lzHw","executionInfo":{"status":"ok","timestamp":1740092807350,"user_tz":360,"elapsed":2920,"user":{"displayName":"Alan Mauricio Camargo Hernandez","userId":"09753390928466343821"}},"outputId":"638835c5-a318-4a91-e60d-b44c886e430e"},"execution_count":26,"outputs":[{"output_type":"stream","name":"stdout","text":["Requirement already satisfied: keras in /usr/local/lib/python3.11/dist-packages (2.0.7)\n","Requirement already satisfied: pyyaml in /usr/local/lib/python3.11/dist-packages (from keras) (6.0.2)\n","Requirement already satisfied: six in /usr/local/lib/python3.11/dist-packages (from keras) (1.17.0)\n","Requirement already satisfied: theano in /usr/local/lib/python3.11/dist-packages (from keras) (1.0.5)\n","Requirement already satisfied: numpy>=1.9.1 in /usr/local/lib/python3.11/dist-packages (from theano->keras) (1.26.4)\n","Requirement already satisfied: scipy>=0.14 in /usr/local/lib/python3.11/dist-packages (from theano->keras) (1.13.1)\n"]}]},{"cell_type":"code","source":["!python -m experiments.random_agent_experiments"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"X9KSkCzilJ6y","executionInfo":{"status":"ok","timestamp":1740092861074,"user_tz":360,"elapsed":3927,"user":{"displayName":"Alan Mauricio Camargo Hernandez","userId":"09753390928466343821"}},"outputId":"1a674f72-8c0e-4c12-9f0a-110915bb6de0"},"execution_count":30,"outputs":[{"output_type":"stream","name":"stdout","text":["Using TensorFlow backend.\n","2025-02-20 23:07:37.648745: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:477] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n","WARNING: All log messages before absl::InitializeLog() is called are written to STDERR\n","E0000 00:00:1740092857.674279    3942 cuda_dnn.cc:8310] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n","E0000 00:00:1740092857.681768    3942 cuda_blas.cc:1418] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n","2025-02-20 23:07:37.705570: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n","To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n","Traceback (most recent call last):\n","  File \"<frozen runpy>\", line 198, in _run_module_as_main\n","  File \"<frozen runpy>\", line 88, in _run_code\n","  File \"/content/emergent_comm_rl/experiments/random_agent_experiments.py\", line 2, in <module>\n","    from data_generator import generate_dummy_categorical_dataset\n","  File \"/content/emergent_comm_rl/data_generator.py\", line 2, in <module>\n","    from keras.utils.np_utils import to_categorical\n","  File \"/usr/local/lib/python3.11/dist-packages/keras/__init__.py\", line 4, in <module>\n","    from . import activations\n","  File \"/usr/local/lib/python3.11/dist-packages/keras/activations.py\", line 6, in <module>\n","    from .engine import Layer\n","  File \"/usr/local/lib/python3.11/dist-packages/keras/engine/__init__.py\", line 8, in <module>\n","    from .training import Model\n","  File \"/usr/local/lib/python3.11/dist-packages/keras/engine/training.py\", line 25, in <module>\n","    from .. import callbacks as cbks\n","  File \"/usr/local/lib/python3.11/dist-packages/keras/callbacks.py\", line 15, in <module>\n","    from collections import Iterable\n","ImportError: cannot import name 'Iterable' from 'collections' (/usr/lib/python3.11/collections/__init__.py)\n"]}]},{"cell_type":"code","source":["python -m experiments.dense_visa_agents_experiments"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":110},"id":"756y1KKjltaD","executionInfo":{"status":"error","timestamp":1740092821519,"user_tz":360,"elapsed":20,"user":{"displayName":"Alan Mauricio Camargo Hernandez","userId":"09753390928466343821"}},"outputId":"e73ca89a-a082-4b7d-b664-04f067670b25"},"execution_count":29,"outputs":[{"output_type":"error","ename":"SyntaxError","evalue":"invalid syntax (<ipython-input-29-0a0b5a539115>, line 1)","traceback":["\u001b[0;36m  File \u001b[0;32m\"<ipython-input-29-0a0b5a539115>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0m\n\u001b[0;31m    python -m experiments.dense_visa_agents_experiments\u001b[0m\n\u001b[0m              ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"]}]}]}